### PR TITLE
Feature/rspec testing

### PR DESCRIPTION
**Pull request by jubaan**
1st rspec review

### CODE REVIEW

**Comment by jubaan** [comment]
The linter error is from another application not from the sticker linter.

File: spec/game_logic_spec.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    end
+  end
+
+  context 'Board' do
+    let(:board) { Board.new }
+    describe '#space_selection' do
+      it 'returns a new board with a changed position' do
+        expect(board.space_selection(4, '+', [])).to eq([4])
+      end
+    end
```
Comment (suretrust) [inline spec/game_logic_spec.rb]: Context should be inside a describe block, not the other way around.  See guidelines [here](https://github.com/rubocop-hq/rspec-style-guide#context-cases).

File: spec/game_logic_spec.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    it 'returns player1 mark' do
+      player1 = Player.new('player1')
+      expect(player1.mark).to eql('+')
+    end
+
+    it 'returns player2 mark' do
+      player2 = Player.new('player2')
+      expect(player2.mark).to eql('-')
+    end
+  end
```
Comment (suretrust) [inline spec/game_logic_spec.rb]: - Specs for each method should be wrapped in one `describe` block.

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi team, 👋 

### Great job thus far! 🙌 Please see my comments on your code. 👇 

- [ ] Ensure your specs cover positive and negative scenarios ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/)) Example: if there is a test for player2 will return player2 mark:
```ruby
it 'returns player1 mark' do
   player1 = Player.new('player1')
   expect(player1.mark).to eql('+')
end
```
There should also be a spec for the negative scenario:
```ruby
it 'returns player1 mark' do
   player1 = Player.new('player1')

   expect(player1.mark).to eql('+')
   expect(player1.mark).not_to eql('/')
end
```

**Kindly submit for another code review request once these issues have been fixed.** _Happy coding!_ 🎉

**Comment by didymus707** [comment]
Testing for Tic Tac Toe: All review changes made

**Comment by didymus707** [comment]
Linter fix caused all tests to fail... All tests are passing

**Review by Forison — APPROVED** [review]
## Status:Approved💚 ,
# Hi@ Team🙌,
 Good job 👏👏 on addressing the changes raised in the previous review.
 The tests look good.
> My opinion on the usage of context: to add to what was said in the previous review. Context is used for more readability of the test. For instance, when you are testing a method that has ie a positive or negative behavior or etc... use context to wrap each scenario that way you have one context of method wrap in one context and the other in another context... When there is one context i don't think it is neccessary to use context
#### Happy coding
#### Addo Forison

**Pull request merged**